fe/fe-core/src/main/java/com/starrocks/scheduler/history/TableKeeper.java

```diff
@@ -62,22 +60,17 @@ public TableKeeper(String database,
 
     public synchronized void run() {
         try {
-            if (!databaseExisted) {
-                databaseExisted = checkDatabaseExists();
-                if (!databaseExisted) {
-                    LOG.warn("database not exists: {}", databaseName);
-                    return;
-                }
+            if (!checkDatabaseExists()) {
+                GlobalStateMgr.getCurrentState().getLocalMetastore().createDb(databaseName);
+                LOG.warn("database created: {}", databaseName);
+                return;
             }
-            if (!tableExisted) {
+            if (!checkTableExists()) {
                 createTable();
                 LOG.info("table created: {}", tableName);
-                tableExisted = checkTableExists();
             }
             correctTable();
-            if (tableExisted) {
-                changeTTL();
-            }
+            changeTTL();
```

Comment:
I don't find any lock on this block. I want to know how to ensure that the table exists when it reaches this point.

Reply:
it will generate an `alter table` sql, using existing sql executor to handle it, which will also check the table existence